libraries/ms-common/src/x86/uart.c

```diff
@@ -1,13 +1,52 @@
 #include "uart.h"
+#include <stdlib.h>
+#include <string.h>
+#include "log.h"
+#include "x86_socket.h"
+
+typedef struct {
+  UARTPort port;
+  X86SocketThread thread;
+  UARTStorage *storage;
+} UartSocketData;
+
+static UartSocketData s_sock[NUM_UART_PORTS];
+
+void uart_receiver_wrapper(struct X86SocketThread *thread, int client_fd, const char *rx_data,
+                           size_t rx_len, void *context);
+
+void uart_receiver_wrapper(struct X86SocketThread *thread, int client_fd, const char *rx_data,
+                           size_t rx_len, void *context) {
+  UartSocketData *sock = context;
+  sock->storage->rx_handler((uint8_t *)rx_data, rx_len, context);
+}
 
 StatusCode uart_init(UARTPort uart, UARTSettings *settings, UARTStorage *storage) {
-  return status_code(STATUS_CODE_UNIMPLEMENTED);
+  // create module name from UART_PORT number
+  size_t module_name_len = 6;
+  char *module_name = malloc(module_name_len);
```

Comment:
Technically this will result in a memory leak, but it's not really a big deal. We don't really have a way to exit cleanly anyways.